### PR TITLE
Misc Middleware cleanup

### PR DESCRIPTION
**Pull request by ErisDS**
A set of small and hopefully un-opinionated cleanup commits, which are a precursor to larger, more opinionated refactors like #7484.
- collecting the two url-related middleware into a package called prettyUrls, will make it easier to include this in multiple places later on, when we treat admin, api & blog as separate apps
- the uncapitalise middleware needed modifications so that it will work when mounted in a sub-app
- ghostlocals was incorrectly lumped in as being theme-specific, when it's actually global middleware that will be needed for all of Ghost
- the ghostVersion logic doesn't need to be in config 🎉 
- static asset handling was littered throughout middleware/index.js making it hard to track, grouping it together helps to determine what belongs before and after assets
- again I've found some static asset handling we aren't using and deleted it

I'm putting these forward as a package of small changes which help lay the ground work for completing #4172.

The remaining work will come in separate PRs 😁  Next: API refactoring

### CODE REVIEW

File: core/server/config/index.js [hunk 1]
```diff
@@ -35,7 +34,6 @@ localUtils.makePathsAbsolute.bind(nconf)();
  * values we have to set manual
  * @TODO: ghost-cli?
  */
-nconf.set('ghostVersion', packageInfo.version);
```
Comment (kirrg001) [inline core/server/config/index.js]: maybe we can remove the `@TODO: ghost-cli?`?
thanks 💯

File: core/server/utils/version.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+var packageInfo = require('../../../package.json'),
```
Comment (kirrg001) [inline core/server/utils/version.js]: suggestion: 
i find the name `version` a bit too general. 
Maybe call it `ghost-version.js`?

Example: I require the file `version  = require('./utils/version')`. When i am reading `verison.full` in the middle of some code, this is hard to know what `version.full` means.

**Review by kirrg001 — COMMENTED** [review]
This all looks great 🌮 
I added two small comments!

**Comment by ErisDS** [comment]
👍 Good calls, fixed up the commit 😁

**Review by kirrg001 — APPROVED** [review]

**Pull request merged**